tck/server.ts

```diff
@@ -0,0 +1,146 @@
+import express from "express";
+import bodyParser from "body-parser";
+import {
+    JSONRPCParams,
+    JSONRPCRequest,
+    JSONRPCServer,
+    SimpleJSONRPCMethod,
+    createJSONRPCErrorResponse,
+} from "json-rpc-2.0";
+
+import * as methods from "./methods";
+import mapping from "./mapping";
+import { RpcMethodParams } from "./params/sdk";
+
+const server = new JSONRPCServer();
+
+// Helper function to register methods
+const registerMethods = (
+    methodsObject: Record<string, string>,
+    prefix = "",
+) => {
+    Object.entries(methodsObject).forEach(([methodName, method]) => {
+        if (methodName === "default" && typeof method === "object") {
+            // Recursively register methods inside the `default` property
+            registerMethods(method, prefix);
+        } else {
+            // Register the method with the correct prefix
+            const fullMethodName = prefix
+                ? `${prefix}.${methodName}`
+                : methodName;
+
+            server.addMethod(
+                fullMethodName,
+                method as unknown as SimpleJSONRPCMethod<void>,
+            );
+        }
+    });
+};
+
+// Register all methods, handling possible nested default export
+if ("default" in methods) {
+    registerMethods(methods.default);
+} else {
+    registerMethods(methods);
+}
+
+// Create mapping server method
+server.addMethod(
+    "mapping",
+    async (...args: [params: RpcMethodParams, serverParams?: void]) => {
+        // Basic mapping for unimplemented functions
+        return await mapping(args[0]);
+    },
+);
+
+/**
+ * Middleware to handle exceptions and customize error responses
+ */
+const exceptionMiddleware = async (
+    next: SimpleJSONRPCMethod,
+    request: JSONRPCRequest,
+    serverParams: JSONRPCParams,
+) => {
+    try {
+        return await next(request, serverParams);
+    } catch (error) {
+        const requestMethod = JSON.stringify(request.method);
+        const errorMessage = error.message || error;
+
+        if (error.status && error.status._code) {
+            console.error(
+                `Hedera error occurred processing JSON-RPC request: ${requestMethod}, Error: ${errorMessage}`,
+            );
+
+            return createJSONRPCErrorResponse(
+                request.id,
+                -32001,
```

Comment:
Can we move this to a constant for better readability. Otherwise looks like a [Magic number](https://en.wikipedia.org/wiki/Magic_number_(programming))